src/ca-table.h

```diff
@@ -285,9 +287,15 @@ class Backend {
   virtual std::unique_ptr<TableBuilder> Create(const char* path,
                                                const TableOptions& options) = 0;
 
-  virtual std::unique_ptr<Table> Open(const char* path) = 0;
+  // NOTE: The fd ownership is transferred to the created object so it is
+  // responsible for closing the file when it is no longer needed.
+  virtual std::unique_ptr<Table> Open(const char* path, int fd,
+                                      const struct stat& st) = 0;
```

Comment:
If you use `kj::AutoCloseFd fd` instead of `int fd`, at least the caller knows they're handing off ownership.  If all they have is an int, they can still do `Open("foo", kj::AutoCloseFd(my_int), st);`